components/tidb_query_normal_expr/src/builtin_string.rs

```diff
@@ -1018,6 +1018,31 @@ impl ScalarFunc {
             .map(|p| p as i64 + 1)
             .or(Some(0)))
     }
+
+    #[inline]
+    pub fn make_set<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let mask = try_opt!(self.children[0].eval_int(ctx, row));
+        let mut output: Vec<u8> = Vec::new();
+        let mut pow2 = 1;
+        let s = b",";
+        let mut q = false;
+        for expr in self.children.iter().skip(1) {
+            if pow2 & mask > 0 {
```

Comment:
TiDB's implementation checks if it's not zero instead of `> 0`. Because if the `bits` are `0xFFFF_FFFF_FFFF_FFFF`, the result can be `< 0`.